### ISSUE TITLE
Gem::UnsatisfiableDependencyError: Unable to resolve dependency: user requested 'bundler (= 1.13.5)'

### ISSUE THREAD

**Issue by bjtieman** [rubygems/rubygems#1915]
I'm having a problem or would like to suggest a feature.

My current problem is when I use gem 2.6.12 I get a "Gem::UnsatisfiableDependencyError" on the bundler gem.  I do not see this error with other versions I've tried (i.e. 2.6.10, 2.5.2)

This issue is related to:

  - [ ] Network problems
  - [ ] Installing a library
  - [ ] Publishing a library
  - [x] The command line `gem`
  - [ ] Other

Here are my current environment details:

```
$ gem env version
2.6.12
```
causes

```
bundler: failed to load command: maestro (/Users/bt023373/Development/gitrepos/mush/vendor/bundle/ruby/2.3.0/bin/maestro)
Gem::UnsatisfiableDependencyError: Unable to resolve dependency: user requested 'bundler (= 1.13.5)'
  /Users/bt023373/.rvm/rubies/ruby-2.3.3/lib/ruby/site_ruby/2.3.0/rubygems/resolver.rb:231:in `search_for'
  /Users/bt023373/.rvm/rubies/ruby-2.3.3/lib/ruby/site_ruby/2.3.0/rubygems/resolver.rb:263:in `block in sort_dependencies'
  /Users/bt023373/.rvm/rubies/ruby-2.3.3/lib/ruby/site_ruby/2.3.0/rubygems/resolver.rb:257:in `each'
  /Users/bt023373/.rvm/rubies/ruby-2.3.3/lib/ruby/site_ruby/2.3.0/rubygems/resolver.rb:257:in `sort_by'
  /Users/bt023373/.rvm/rubies/ruby-2.3.3/lib/ruby/site_ruby/2.3.0/rubygems/resolver.rb:257:in `sort_dependencies'
  /Users/bt023373/.rvm/rubies/ruby-2.3.3/lib/ruby/site_ruby/2.3.0/rubygems/resolver/molinillo/lib/molinillo/delegates/specification_provider.rb:52:in `block in sort_dependencies'
  /Users/bt023373/.rvm/rubies/ruby-2.3.3/lib/ruby/site_ruby/2.3.0/rubygems/resolver/molinillo/lib/molinillo/delegates/specification_provider.rb:69:in `with_no_such_dependency_error_handling'
  /Users/bt023373/.rvm/rubies/ruby-2.3.3/lib/ruby/site_ruby/2.3.0/rubygems/resolver/molinillo/lib/molinillo/delegates/specification_provider.rb:51:in `sort_dependencies'
  /Users/bt023373/.rvm/rubies/ruby-2.3.3/lib/ruby/site_ruby/2.3.0/rubygems/resolver/molinillo/lib/molinillo/resolution.rb:165:in `initial_state'
  /Users/bt023373/.rvm/rubies/ruby-2.3.3/lib/ruby/site_ruby/2.3.0/rubygems/resolver/molinillo/lib/molinillo/resolution.rb:106:in `start_resolution'
  /Users/bt023373/.rvm/rubies/ruby-2.3.3/lib/ruby/site_ruby/2.3.0/rubygems/resolver/molinillo/lib/molinillo/resolution.rb:64:in `resolve'
  /Users/bt023373/.rvm/rubies/ruby-2.3.3/lib/ruby/site_ruby/2.3.0/rubygems/resolver/molinillo/lib/molinillo/resolver.rb:42:in `resolve'
  /Users/bt023373/.rvm/rubies/ruby-2.3.3/lib/ruby/site_ruby/2.3.0/rubygems/resolver.rb:188:in `resolve'
  /Users/bt023373/.rvm/rubies/ruby-2.3.3/lib/ruby/site_ruby/2.3.0/rubygems/request_set.rb:368:in `resolve'
  /Users/bt023373/.rvm/rubies/ruby-2.3.3/lib/ruby/site_ruby/2.3.0/rubygems/request_set.rb:380:in `resolve_current'
  /Users/bt023373/.rvm/rubies/ruby-2.3.3/lib/ruby/site_ruby/2.3.0/rubygems.rb:239:in `finish_resolve'
  /Users/bt023373/.rvm/rubies/ruby-2.3.3/lib/ruby/site_ruby/2.3.0/rubygems/rdoc.rb:15:in `<top (required)>'
  /Users/bt023373/.rvm/rubies/ruby-2.3.3/lib/ruby/site_ruby/2.3.0/rubygems/core_ext/kernel_require.rb:55:in `require'
  /Users/bt023373/.rvm/rubies/ruby-2.3.3/lib/ruby/site_ruby/2.3.0/rubygems/core_ext/kernel_require.rb:55:in `require'
  /Users/bt023373/.rvm/rubies/ruby-2.3.3/lib/ruby/site_ruby/2.3.0/rubygems/uninstaller.rb:11:in `<top (required)>'
  /Users/bt023373/.rvm/rubies/ruby-2.3.3/lib/ruby/site_ruby/2.3.0/rubygems/core_ext/kernel_require.rb:55:in `require'
  /Users/bt023373/.rvm/rubies/ruby-2.3.3/lib/ruby/site_ruby/2.3.0/rubygems/core_ext/kernel_require.rb:55:in `require'
...
```

but when I downgrade to 2.6.10 or even 2.5.2 I do not get the same error.  I haven't tried other versions.

I will abide by the [code of conduct](https://github.com/rubygems/rubygems/blob/master/CODE_OF_CONDUCT.md).

**Comment by segiddins** [comment]
What command are you running that's failing? It looks like you're trying to `bundle exec`, what's the output of `bundle env`?

**Comment by bjtieman** [comment]
Bundle 1.14.6

> bundle env
## Environment

```
Bundler   1.14.6
Rubygems  2.6.12
Ruby      2.3.3p222 (2016-11-21 revision 56859) [x86_64-darwin15]
GEM_HOME  /Users/bt023373/.rvm/gems/ruby-2.3.3
GEM_PATH  /Users/bt023373/.rvm/gems/ruby-2.3.3:/Users/bt023373/.rvm/gems/ruby-2.3.3@global
RVM       1.29.1 (latest)
Git       2.10.1
Platform  x86_64-darwin-15
rubygems-bundler (1.4.4)
```

## Bundler settings

```
gem.coc
  Set for the current user (/Users/bt023373/.bundle/config): "true"
gem.mit
  Set for the current user (/Users/bt023373/.bundle/config): "false"
gem.test
  Set for the current user (/Users/bt023373/.bundle/config): "rspec"
path
  Set for your local app (/Users/bt023373/Development/gitrepos/mush/.bundle/config): "vendor/bundle"
disable_shared_gems
  Set for your local app (/Users/bt023373/Development/gitrepos/mush/.bundle/config): "true"
```

## Gemfile

### Gemfile

```ruby
source 'http://rubygems.org'
source 'http://repo.release.cerner.corp/main/rubygems/'
source 'http://repo.release.cerner.corp/internal/rubygems/'

source 'http://rubygems.org/' do
  gem 'chef', '~> 12.6'
  gem 'english'
  gem 'rake', '~> 10.4'
end

source 'http://repo.release.cerner.corp/main/rubygems/' do
  gem 'timber'

  gem 'maestro', '~> 0.10'
  gem 'saddle', '~> 0.5'
  gem 'wolfe-manager-client-ruby', '~> 2.6'
end

group 'testing' do
  gem 'rspec', '~> 3.4'
end
```

### Gemfile.lock

```
GEM
  remote: http://repo.release.cerner.corp/internal/rubygems/
  remote: http://rubygems.org/
  remote: http://repo.release.cerner.corp/main/rubygems/
  specs:
    activesupport (4.2.8)
      i18n (~> 0.7)
      minitest (~> 5.1)
      thread_safe (~> 0.3, >= 0.3.4)
      tzinfo (~> 1.1)
    addressable (2.5.1)
      public_suffix (~> 2.0, >= 2.0.2)
    axiom-types (0.1.1)
      descendants_tracker (~> 0.0.4)
      ice_nine (~> 0.11.0)
      thread_safe (~> 0.3, >= 0.3.1)
    buff-config (2.0.0)
      buff-extensions (~> 2.0)
      varia_model (~> 0.6)
    buff-extensions (2.0.0)
    buff-ignore (1.2.0)
    buff-ruby_engine (1.0.0)
    buff-shell_out (1.1.0)
      buff-ruby_engine (~> 1.0)
    builder (3.2.3)
    celluloid (0.16.0)
      timers (~> 4.0.0)
    celluloid-io (0.16.2)
      celluloid (>= 0.16.0)
      nio4r (>= 1.1.0)
    cerner_mustard (0.2.2)
      ridley (~> 5.0)
    chef (12.20.3)
      addressable
      bundler (>= 1.10)
      chef-config (= 12.20.3)
      chef-zero (>= 4.8)
      diff-lcs (~> 1.2, >= 1.2.4)
      erubis (~> 2.7)
      ffi-yajl (~> 2.2)
      highline (~> 1.6, >= 1.6.9)
      iniparse (~> 1.4)
      mixlib-archive (~> 0.4)
      mixlib-authentication (~> 1.4)
      mixlib-cli (~> 1.7)
      mixlib-log (~> 1.3)
      mixlib-shellout (~> 2.0)
      net-sftp (~> 2.1, >= 2.1.2)
      net-ssh (>= 2.9, < 5.0)
      net-ssh-multi (~> 1.2, >= 1.2.1)
      ohai (>= 8.6.0.alpha.1, < 13)
      plist (~> 3.2)
      proxifier (~> 1.0)
      rspec-core (~> 3.5)
      rspec-expectations (~> 3.5)
      rspec-mocks (~> 3.5)
      rspec_junit_formatter (~> 0.2.0)
      serverspec (~> 2.7)
      specinfra (~> 2.10)
      syslog-logger (~> 1.6)
      uuidtools (~> 2.1.5)
    chef-config (12.20.3)
      addressable
      fuzzyurl
      mixlib-config (~> 2.0)
      mixlib-shellout (~> 2.0)
    chef-zero (13.0.0)
      ffi-yajl (~> 2.2)
      hashie (>= 2.0, < 4.0)
      mixlib-log (~> 1.3)
      rack (~> 2.0)
      uuidtools (~> 2.1)
    coercible (1.0.0)
      descendants_tracker (~> 0.0.1)
    colorize (0.5.8)
    concurrent-ruby (0.7.1)
      ref (~> 1.0.5)
    descendants_tracker (0.0.4)
      thread_safe (~> 0.3, >= 0.3.1)
    diff-lcs (1.3)
    english (0.6.3)
      language
    equalizer (0.0.11)
    erubis (2.7.0)
    faraday (0.9.2)
      multipart-post (>= 1.2, < 3)
    faraday_middleware (0.11.0.1)
      faraday (>= 0.7.4, < 1.0)
    ffi (1.9.18)
    ffi-yajl (2.3.0)
      libyajl2 (~> 1.2)
    fuzzyurl (0.9.0)
    git (1.2.8)
    hashie (3.5.5)
    highline (1.7.8)
    hitimes (1.2.4)
    httpclient (2.8.3)
    i18n (0.8.1)
    ice_nine (0.11.2)
    iniparse (1.4.2)
    ipaddress (0.8.3)
    json (1.8.6)
    knife-zabbix (3.0.1)
      nokogiri (~> 1.6.7)
      rb-readline
      zabbixapi (~> 2.4.7)
    language (0.6.0)
    libyajl2 (1.2.0)
    maestro (0.16.0)
      colorize (= 0.5.8)
      concurrent-ruby (= 0.7.1)
      git (= 1.2.8)
      rash-cli (= 1.0.0)
    mini_portile2 (2.1.0)
    minitest (5.10.1)
    mixlib-archive (0.4.1)
      mixlib-log
    mixlib-authentication (1.4.1)
      mixlib-log
    mixlib-cli (1.7.0)
    mixlib-config (2.2.4)
    mixlib-log (1.7.1)
    mixlib-shellout (2.2.7)
    multi_json (1.12.1)
    multipart-post (2.0.0)
    net-scp (1.2.1)
      net-ssh (>= 2.6.5)
    net-sftp (2.1.2)
      net-ssh (>= 2.6.5)
    net-ssh (4.1.0)
    net-ssh-gateway (2.0.0)
      net-ssh (>= 4.0.0)
    net-ssh-multi (1.2.1)
      net-ssh (>= 2.6.5)
      net-ssh-gateway (>= 1.2.0)
    net-telnet (0.1.1)
    nio4r (2.0.0)
    nokogiri (1.6.8.1)
      mini_portile2 (~> 2.1.0)
    ohai (8.23.0)
      chef-config (>= 12.5.0.alpha.1, < 13)
      ffi (~> 1.9)
      ffi-yajl (~> 2.2)
      ipaddress
      mixlib-cli
      mixlib-config (~> 2.0)
      mixlib-log (>= 1.7.1, < 2.0)
      mixlib-shellout (~> 2.0)
      plist (~> 3.1)
      systemu (~> 2.6.4)
      wmi-lite (~> 1.0)
    plist (3.3.0)
    proxifier (1.0.3)
    public_suffix (2.0.5)
    rack (2.0.2)
    rake (10.5.0)
    rash-cli (1.0.0)
      colorize (= 0.5.8)
    rb-readline (0.5.4)
    ref (1.0.5)
    retryable (2.0.4)
    ridley (5.1.0)
      addressable
      buff-config (~> 2.0)
      buff-extensions (~> 2.0)
      buff-ignore (~> 1.2)
      buff-shell_out (~> 1.0)
      celluloid (~> 0.16.0)
      celluloid-io (~> 0.16.1)
      chef-config (>= 12.5.0)
      erubis
      faraday (~> 0.9.0)
      hashie (>= 2.0.2, < 4.0.0)
      httpclient (~> 2.7)
      json (>= 1.7.7)
      mixlib-authentication (>= 1.3.0)
      retryable (~> 2.0)
      semverse (~> 2.0)
      varia_model (~> 0.6)
    rolling_pin (0.1.1)
      cerner_mustard (~> 0.1)
      chef (>= 12.0)
      thor (~> 0.19)
    rspec (3.6.0)
      rspec-core (~> 3.6.0)
      rspec-expectations (~> 3.6.0)
      rspec-mocks (~> 3.6.0)
    rspec-core (3.6.0)
      rspec-support (~> 3.6.0)
    rspec-expectations (3.6.0)
      diff-lcs (>= 1.2.0, < 2.0)
      rspec-support (~> 3.6.0)
    rspec-its (1.2.0)
      rspec-core (>= 3.0.0)
      rspec-expectations (>= 3.0.0)
    rspec-mocks (3.6.0)
      diff-lcs (>= 1.2.0, < 2.0)
      rspec-support (~> 3.6.0)
    rspec-support (3.6.0)
    rspec_junit_formatter (0.2.3)
      builder (< 4)
      rspec-core (>= 2, < 4, != 2.12.0)
    saddle (0.5.0)
      cerner_mustard (~> 0.1)
      chef (~> 12.6)
      git (~> 1.2)
      knife-zabbix (~> 3.0)
      maestro (~> 0.10)
      rolling_pin (~> 0.1)
      zabbixapi (~> 2.4.7)
    semverse (2.0.0)
    serverspec (2.38.0)
      multi_json
      rspec (~> 3.0)
      rspec-its
      specinfra (~> 2.53)
    sfl (2.3)
    specinfra (2.67.9)
      net-scp
      net-ssh (>= 2.7, < 5.0)
      net-telnet
      sfl
    syslog-logger (1.6.8)
    systemu (2.6.5)
    thor (0.19.4)
    thread_safe (0.3.6)
    timber (1.9.0)
      activesupport
    timers (4.0.4)
      hitimes
    tzinfo (1.2.3)
      thread_safe (~> 0.1)
    uuidtools (2.1.5)
    varia_model (0.6.0)
      buff-extensions (~> 2.0)
      hashie (>= 2.0.2, < 4.0.0)
    virtus (1.0.5)
      axiom-types (~> 0.1)
      coercible (~> 1.0)
      descendants_tracker (~> 0.0, >= 0.0.3)
      equalizer (~> 0.0, >= 0.0.9)
    virtus_convert (0.1.0)
    wmi-lite (1.0.0)
    wolfe-manager-client-ruby (2.8.0)
      faraday (~> 0.8)
      faraday_middleware (~> 0.8)
      timber (~> 1.3)
      virtus (~> 1.0)
      virtus_convert (~> 0.1)
    zabbixapi (2.4.9)
      json (~> 1.6, >= 1.6.0)

PLATFORMS
  ruby

DEPENDENCIES
  chef (~> 12.6)!
  english!
  maestro (~> 0.10)!
  rake (~> 10.4)!
  rspec (~> 3.4)
  saddle (~> 0.5)!
  timber!
  wolfe-manager-client-ruby (~> 2.6)!

BUNDLED WITH
   1.13.5
```

The error occurs when we try to bundle exec a command in the project.

**Comment by segiddins** [comment]
I believe this issue should be resolved with Bundler 1.15.0.pre.3

**Comment by bjtieman** [comment]
Yes, it does look like 1.15.0.pre.3 does not have this issue.

Thank you for your time...closing...
